spec/rewrite/client_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+describe EasyPost::Client do
+  describe 'client object' do
+    it 'create a client object timeouts' do
+      client = described_class.new(api_key: 'fake_api_key', read_timeout: 10, open_timeout: 1)
+
+      expect(client.read_timeout).to eq(10)
+      expect(client.open_timeout).to eq(1)
+    end
+
+    it 'create a client with api key and api base' do
+      client = described_class.new(api_key: 'fake_api_key', api_base: 'https://api.easypostExample.com')
+      expect(client.api_base).to eq('https://api.easypostExample.com')
+    end
+
+    it 'create a client with a missing api key' do
+      expect {
+        described_class.new(api_key: nil)
+      }.to raise_error(EasyPost::Error, 'API key is required.')
+    end
+
+    it 'create a client with read timeout' do
+      client = described_class.new(api_key: ENV['EASYPOST_TEST_API_KEY'], read_timeout: 0.001, open_timeout: 10)
+
+      expect {
+        client.address.create(Fixture.ca_address1)
+      }.to raise_error(Net::ReadTimeout)
```

Comment:
This should raise our EasyPost timeout error

I guess we haven't overhauled the error handling yet right? Can you leave a TODO to update the test to use our EasyPost timeout error when we get there?

Reply:
I will add a TODO in the part two PR, can I get a stamp 😉

Reply:
I mean, you could also address this and then got a stamp from me seeing as I haven't stamped it either way 😉

Reply:
Doesn't want to dismiss Nate's stamp😉, adding this to part 2 now as we speak so I don't forget. Thanks for the approval!

Reply:
It would be permisseble to dismiss the stamp since he already approved and all you're doing is adding a comment. Once I approved you could merge